alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -291,7 +292,11 @@ object MonitorRunner : JobRunner, CoroutineScope, AbstractLifecycleComponent() {
 
             if (triggerService.isQueryLevelTriggerActionable(triggerCtx, triggerResult)) {
                 val actionCtx = triggerCtx.copy(error = monitorResult.error ?: triggerResult.error)
-                for (action in trigger.actions) {
+                var threshold = ALERTING_TRIGGER_MAX_ACTIONS.get(settings)
+                threshold = if (threshold < trigger.actions.size) threshold else trigger.actions.size
+
+                // val threshold = if(ALERTING_TRIGGER_MAX_ACTIONS.get(settings) < trigger.actions.size) ALERTING_TRIGGER_MAX_ACTIONS.get(settings) else trigger.actions.size
+                for (action in trigger.actions.slice(0 until threshold)) {
```

Comment:
Since the default value for `ALERTING_TRIGGER_MAX_ACTIONS` is `-1` won't this slice just return nothing meaning no actions will be evaluated?

I'm assuming the default behavior we want is to evaluate all actions when the setting is `-1` so this logic will need to account for that.

You'll also want to add the logic to implement `ALERTING_TRIGGER_MAX_ACTIONS` for Bucket-Level Monitors.

Also, let's add some tests to cover the behavior of this new setting. If you want to add integration tests for Monitor execution behavior, you can write it in [MonitorRunnerIT](https://github.com/opensearch-project/alerting/blob/main/alerting/src/test/kotlin/org/opensearch/alerting/MonitorRunnerIT.kt)

Reply:
> Since the default value for `ALERTING_TRIGGER_MAX_ACTIONS` is `-1` won't this slice just return nothing meaning no actions will be evaluated?
> 
> I'm assuming the default behavior we want is to evaluate all actions when the setting is `-1` so this logic will need to account for that.
> Got it! I added `&& threshold >= 0` right after `threshold < trigger.actions.size`. how is that sounds like or i will still have some logical problem?
> You'll also want to add the logic to implement `ALERTING_TRIGGER_MAX_ACTIONS` for Bucket-Level Monitors.
> Got it! So basically, what I'm going to do is add the same thing like what i did in the Query Level right?
> Also, let's add some tests to cover the behavior of this new setting. If you want to add integration tests for Monitor execution behavior, you can write it in [MonitorRunnerIT](https://github.com/opensearch-project/alerting/blob/main/alerting/src/test/kotlin/org/opensearch/alerting/MonitorRunnerIT.kt)
I tested it in the OpensearchDashboard. Do i still need to add integration tests for Monitor execution behavior?